backend/postgres/postgres.go

```diff
@@ -0,0 +1,856 @@
+package postgresbackend
+
+import (
+	"context"
+	"database/sql"
+	"embed"
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log/slog"
+	"strings"
+	"time"
+
+	"github.com/cschleiden/go-workflows/backend"
+	"github.com/cschleiden/go-workflows/backend/converter"
+	"github.com/cschleiden/go-workflows/backend/history"
+	"github.com/cschleiden/go-workflows/backend/metadata"
+	"github.com/cschleiden/go-workflows/backend/metrics"
+	"github.com/cschleiden/go-workflows/core"
+	"github.com/cschleiden/go-workflows/internal/metrickeys"
+	"github.com/cschleiden/go-workflows/internal/workflowerrors"
+	"github.com/cschleiden/go-workflows/workflow"
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/golang-migrate/migrate/v4/database/postgres"
+	"github.com/golang-migrate/migrate/v4/source/iofs"
+	"github.com/google/uuid"
+	_ "github.com/lib/pq"
+	"go.opentelemetry.io/otel/trace"
+)
+
+//go:embed db/migrations/*.sql
+var migrationsFS embed.FS
+
+func SQLReplacer(src string) string {
+	for nParam := 1; strings.Contains(src, "?"); nParam++ {
+		src = strings.Replace(src, "?", fmt.Sprintf("$%d", nParam), 1)
+	}
+	return src
+}
```

Comment:
Is that the only change in the code compared to the mysql backend or was there more you had to change?

Reply:
If I can remember it correct, the migration sql files and the SQLReplacer was the only thing i changed in order to get postgres to work. Wanted the query's to remain as same as possible so there would be only copy-paste without much thinking to do.

Reply:
there are minor changes in backend/postgres/postgres.go file:

dsn := fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=require", host, port, user, password, database)

db, err := sql.Open("postgres", dsn)

db, err := sql.Open("postgres", schemaDsn)

dbi, err := postgres.WithInstance(db, &postgres.Config{})


m, err := migrate.NewWithInstance("iofs", migrations, "postgres", dbi)